test-harness/src/main/java/edu/umd/cs/findbugs/test/matcher/BugInstanceMatcher.java

```diff
@@ -109,8 +109,8 @@ public boolean matches(Object obj) {
                 String fullName = classAnn.getClassName();
                 int startDot = fullName.lastIndexOf(".") + 1;
                 int endDollar = fullName.indexOf('$');
-                String simpleName = fullName.substring(startDot != -1 ? startDot : 0, endDollar != -1 ? endDollar : fullName.length());
-                String simpleNameInner = fullName.substring(startDot != -1 ? startDot : 0, fullName.length());
+                String simpleName = fullName.substring(startDot, endDollar != -1 ? endDollar : fullName.length());
+                String simpleNameInner = fullName.substring(startDot);
```

Comment:
The value of `startDot` can never be -1, since the smallest value of `lastIndexOf()` is -1, but there is a +1.